### PR TITLE
fix: migrated langchain_rag example to latest langchain and openai libs

### LINKED ISSUE #77: Migrate langchain_rag example to the latest version of langchain

### PR DESCRIPTION
**Pull request by adubovik**
Resolves https://github.com/epam/ai-dial-sdk/issues/77

### CODE REVIEW

File: examples/langchain_rag/.env.example [hunk 1]
```diff
@@ -0,0 +1,9 @@
+DIAL_URL=DIAL_URL
+
+CHAT_MODEL=gpt-4
+EMBEDDINGS_MODEL=text-embedding-ada-002
+API_VERSION=2023-03-15-preview
```
Comment (vladisavvv) [inline examples/langchain_rag/.env.example]: Let's change to the latest stable

File: examples/langchain_rag/app.py [hunk 2]
```diff
@@ -33,127 +55,136 @@ class CustomCallbackHandler(AsyncCallbackHandler):
...
-                ),
-            )
-
-            embeddings = CacheBackedEmbeddings.from_bytes_store(
-                openai_embedding,
-                embedding_store,
-                namespace=openai_embedding.model,
-            )
+            file_url = get_last_attachment_url(request.messages)
+            file_abs_url = urljoin(f"{DIAL_URL}/v1/", file_url)
```
Comment (vladisavvv) [inline examples/langchain_rag/app.py:67]: Previously, the application worked with links to websites. Why do we want to change this?

File: examples/langchain_rag/.env.example (hunk 1 above)
Comment (adubovik) [reply examples/langchain_rag/.env.example]: Fixed. Used `2024-02-01` as per [doc](https://learn.microsoft.com/en-us/azure/ai-services/openai/reference#chat-completions).

File: examples/langchain_rag/app.py (hunk 2 above)
Comment (adubovik) [reply examples/langchain_rag/app.py:67]: It still works with links to websites.

**Review by vladisavvv — DISMISSED** [review]

**Review by vladisavvv — DISMISSED** [review]

**Review by vladisavvv — APPROVED** [review]

**Pull request merged**